File: tests/test_graph_cut.rs
use gpf::graphcut::{MaxFlow, PushRelabelFifo};

#[test]
fn test_max_flow() {
    let arcs = vec![
        (0, 1, 10),
        (0, 2, 6),
        (1, 3, 4),
        (3, 1, 4),
        (1, 4, 4),
        (4, 1, 4),
        (2, 5, 4),
        (5, 2, 4),
        (3, 4, 2),
        (4, 3, 2),
        (4, 5, 2),
        (5, 4, 2),
        (3, 6, 6),
        (5, 6, 6),
    ];
    let mut graph = PushRelabelFifo::from((arcs, 7));
    graph.find_max_flow();

    println!("haha");
}

File: tests/test_graphcut.cpp
#include "gpf/graphcut.hpp"
#include <iostream>
#include <cassert>
#include <cmath>

int main() {
    std::vector<double> source_cap = {10.0, 5.0, 15.0};
    std::vector<double> sink_cap = {8.0, 10.0, 7.0};
    
    gpf::GraphCut gc(source_cap, sink_cap);
    
    gc.add_edge(0, 1, 10.0, 10.0);
    gc.add_edge(1, 2, 5.0, 5.0);
    gc.add_edge(0, 2, 15.0, 15.0);
    
    double flow = gc.max_flow();
    
    assert(flow > 0.0);
    std::cout << "Max flow: " << flow << "\n";
    std::cout << "Graph cut tests passed!\n";
    return 0;
}
